Add tsc-generated type definitions for opt-in (#12431)
openlayers/openlayers: README.md

See our [GitHub sponsors page](https://github.com/sponsors/openlayers) or [Open Collective](https://opencollective.com/openlayers/contribute/sponsors-214/checkout) if you too are interested in becoming a regular sponsor.

> ## Building ES6

Reviewer: Pretty petty issue, but the ECMAScript 6th edition (ES6) was never published.  Instead it was renamed to ES2015.  Rather than pinning ourselves to a specific version, I think we could use "ES Modules."

And then instead of "Building ES6," since this section is about using the untranspiled sources directly, maybe the heading could be "Using ES Modules" or just "ES Modules."

Reviewer: ```suggestion
## ES Modules
```

> The `ol` package contains a `src/` folder with the sources, authored in ES6. To use these untranspiled sources, either import modules from `ol/src` instead of `ol`, or configure your bundler with an alias pointing to `ol/src` for the `ol` package.

Reviewer: ```suggestion
The `ol` package contains a `src/` folder with the sources, authored as [ES Modules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules). To use these untranspiled sources, either import modules from `ol/src` instead of `ol`, or configure your bundler with an alias pointing to `ol/src` for the `ol` package.
```

## IntelliSense (VS Code) support

> The untranspiled sources in the `src/` are JSDoc type annotated. For applications authored in JavaScript, VS Code can get type definitions from these sources with a properly configured `jsconfig.json` in the project root:

Reviewer 2: ```suggestion
The untranspiled sources in the `src/` folder are JSDoc type annotated. For applications authored in JavaScript, VS Code can get type definitions from these sources with a properly configured `jsconfig.json` in the project root:
```

  }
}
```
</details>

## TypeScript support

When authoring in TypeScript, we recommend you try out the types that are included in the `ol` package. To use these types, make sure you have the following entry for `@types/ol` in your `package.json`'s `devDependencies` section:
```
{
  ...
  "devDependencies": {
    ...
    "@types/ol": "file:node_modules/ol/types",
    ...
  }
}
```

> These are auto-generated with the TypeScript compiler, and will be the default in future versions. Alternatively, you can use third-party types from [Definitely Typed](https://definitelytyped.org) (`npm install @types/ol`).

Reviewer 2: Last time I checked (maybe around december last year) the `@hanreev/types-ol` were providing better typing then the `@types/ol`. Maybe mentioning both?

Reviewer 2: ```suggestion
These are auto-generated with the TypeScript compiler, and will be the default in future versions. Alternatively, you can use third-party types from [Definitely Typed](https://definitelytyped.org) (`npm install @types/ol`) or from [hanreev/types-ol](https://github.com/hanreev/types-ol).
```
